README.md
# cr-tax

react component tax and bill styles

## Example View
<img src="http://39.107.35.212/images/cr-tax/example2.png" alt="example" />

## All TaxTemplate

- 增值税专用发票
- 增值税普通发票
- 增值税电子普通发票
- 机动车销售统一发票
- 增值税普通发票（卷票）
- 国税通用机打发票
- 增值税电子普通发票（通行费）
- 海关进口增值税专用缴费书
- 无票收入
- 二手车销售统一发票
- 增值税电子普通发票（旅客运输）
- 增值税电子普通发票农产品

## Usage

```bash
cnpm install --save cr-tax
# or
npm install --save cr-tax
```

```js
import React, { PureComponent } from 'react';
import Tax from 'cr-tax';
import 'cr-tax/assets/index.css';

const taxHeader = {
  number: '2897454',
  code: '4646456',
  date: '20190911',
}
const taxPurchase = {
  name: '南京智慧科技有限公司',
  taxNumber: '142704654502922440',
  addressPhone: '南京南路 123456',
  bank: '',
}
const taxSales = {
  totalB: '贰仟零壹拾叁元柒角陆分',
  totalS: '¥2,013.76',
  name: '衡水华业工业有限公司',
  taxNumber: '789754654444464',
  addressPhone: '运城市运万路幸福庄0350-7878505',
  bank: '农行运城北城支行56600104454502242',
  remark: '',
}
const taxDataList = [{
  name: '*钢件*电池阴极金具',
  type: 'ER602EGK6001',
  unit: '个',
  number: '3,000.00',
  price: '3,000.00',
  sum: '3,000.00',
  rate: 0.16,
  tax: '435435.16',
}]

class AppB extends PureComponent {
  render () {
    return (
      <Tax
        mode={'ZZSZYFP'}
        taxHeader={taxHeader}
        taxPurchase={taxPurchase}
        ticketType={1}
        taxDataList={taxDataList}
        taxSales={taxSales}
      />
    )
  }
}
```

## API

### 增值税专用发票

name | type | description
-----| -----| ------------
mode | string | ZZSZYFP
taxHeader | object | 发票头部信息 { code: 发票代码; number: 发票号码; date: 开票日期; identiDate: 认证日期;  }
taxPurchase | object | 购货单位信息 { name: *名称; taxNumber: 纳税人识别号; addressPhone: 地址、电话; bank: 开户行及账户 }
taxDataList | array[object] | 货物数据列表 [{ name: *货物或应税劳务名称; type: 规格型号; unit: 单位; number: 数量; price: 单价; sum: *金额; rate: *税率; tax: 税额 }]
taxSales | object | 销货单位及价税合计信息 { totalB: 价税合计(大写); totalS: (小写); name: *名称; taxNumber: 纳税人识别号; addressPhone: 地址、电话; bank: 开户行及账户; remark: 备注 }
ticketType | number | 开票类型 1: 代开发票; 2: 自开发票
direction | string(oneof ['in', 'out']) | in: 进项发票; out: 销项发票

### 增值税电子普通发票

name | type | description
-----| -----| ------------
mode | string | ZZSDZPTFP
taxHeader | object | 发票头部信息 { code: 发票代码; number: 发票号码; date: 开票日期; identiDate: 认证日期; }
taxPurchase | object | 购货单位信息 { name: *名称; taxNumber: 纳税人识别号; addressPhone: 地址、电话; bank: 开户行及账户 }
taxDataList | array[object] | 货物数据列表 [{ name: *项目名称; carNumber: 车牌号; type: 类型; dateStart: 通行日期起; dateEnd: 通行日期止; sum: *金额; rate: *税率; tax: 税额 }]
taxSales | object | 销货单位及价税合计信息 { totalB: 价税合计(大写); totalS: (小写); name: *名称; taxNumber: 纳税人识别号; addressPhone: 地址、电话; bank: 开户行及账户; remark: 备注 }
ticketType | number | 开票类型 1: 代开发票; 2: 自开发票
direction | string(oneof ['in', 'out']) | in: 进项发票; out: 销项发票

### 机动车销售统一发票

name | type | description
-----| -----| ------------
mode | string | JDCXSTYFP
prefixCls | string | cr-tax-c
taxHeader | object | 发票头部信息 { code: 发票代码; number: 发票号码; date: 开票日期 }
mainData | object | 发票主体信息 { machineCode: 机打代码; machineNumber: 机打号码; encipher: 机器编码; purchase:  *购买方名称及身份证号码/组织机构代码; purchaseTaxNumber: 纳税人识别号(购买方); carType: 车辆类型; factory: 厂牌型号; origin: *产地; certificate: 合格证号; importNumber: 进口证明书号; inspectionNumber: 商检单号; engineNumber: 发动机号码; carNumber: 车牌识别号/车架号码; totalB: 价税合计（大写）; totalS: *小写; sales: *销货单位名称; phone: 电话; salesTaxNumber: 纳税人识别号(销货方); account: 账号; address: 地址; bank: 开户银行; rate: *增值税税率或征收率; tax: *增值税税额; chargeTaxCode: 主管税务机关及代码; notTaxCount: *不含税价; taxCertificate: 完税凭证号码; weight: 吨位; maxNumber: 限乘人数 }

## Development

```
cnpm install
npm start
```

## Example

http://localhost:8018/examples/testOne.html

## Test
  ```js
  npm test
  
  npm run coverage
  ```


## License

cr-tax is released under the MIT license.


HISTORY.md
# History
----

## 1.4.2 / 2020-01-13
增加 增值税电子普通发票农产品 发票

## 1.4.0
添加 zf-tools 

## 1.3.9
修改国税通用机打发票接口字段

## 1.3.7
修改发票数据结构

## 1.3.6
修改机动车销售统一发票数据结构

## 1.3.5
增加电子普通发票（旅客运输）

## 1.3.4
增加海关进口专用缴费书
增加无票收入
修改专票、普票、卷票数据结构

## 1.3.3
增加增值税电子普通发票(通行费)

## 1.2.4
修改样式问题

## 1.1.7
解决数据过多出现不对齐现象

## 1.1.2 / 2019-11-04
增加增值税普通发票
价税合计数字格式化显示

## 1.0.8 / 2019-09-26
数据列表过多，增加纵向滚动条

## 1.0.5 / 2019-09-23
数据防护

## 1.0.4 / 2019-09-19
增加统一说明文档

## 1.0.3 / 2019-09-17
增加机动车销售统一发票

## 1.0.2 / 2019-09-16
增加增值税电子通用发票

## 1.0.0 / 2019-04-13

comta is start



examples/testOne.js
import React from 'react';
import ReactDOM from 'react-dom';
import Testone from 'cr-tax';
import '../assets/index.less';
import { GSTYJD, PTFPJP, HGJKZZS, WPSR, ESCXS, LKYS, NCP } from 'cr-tax';

const reactContainer = document.getElementById('__react-content');
const bodyContainer = document.getElementsByTagName('body')
bodyContainer[0].style.padding = '10px'
reactContainer.style.cssText = `
                                // border: 1px solid #11d0bc;
                                // padding: 10px;
                                // margin-bottom: 10px;
                               `
const boxStyle = {
  // width: 1150,
  // height: 'auto',
  // margin: '0 auto',
  // border: '1px solid green',
}

const taxHeader = {
  number: '2897454',
  code: '4646456',
  date: '20190911',
}
const taxPurchase = {
  name: '南京智慧科技有限公司',
  taxNumber: '142704654502922440',
  addressPhone: '南京南路 123456',
  bank: '',
}
const taxSales = {
  totalB: '贰仟零壹拾叁元柒角陆分',
  totalS: '2013.76',
  name: '衡水华业工业有限公司',
  taxNumber: '789754654444464',
  addressPhone: '运城市运万路幸福庄0350-7878505',
  bank: '农行运城北城支行56600104454502242',
  remark: '',
}
const taxSalesB = {
  totalB: '贰仟零壹拾叁元柒角陆分',
  totalS: '2013.76',
  name: '衡水华业工业有限公司',
  taxNumber: '789754654444464',
  addressPhone: '运城市运万路幸福庄0350-44520505',
  bank: '农行运城北城支行56600104545002242',
  remark: '',
}
const taxDataList = [{
  name: '*钢件*电池运城市运万路幸福庄0350-44520505',
  billNumber: 'ER602EGK6001',
  modelUnit: '个',
  modelCount: '3',
  modelPrice: '3000',
  price: '56125489',
  taxRate: 0.17,
  billTaxValue: '435435.16',
}, {}, {}]

const taxDataListB = [{
  name: '*钢件*电池运城市运万路幸福庄0350-44520505',
  modelNumber: 'ER602EGK6001',
  modelUnit: '个',
  modelCount: '20180108',
  modelPrice: '20180108',
  price: '3,000.00',
  taxRate: 0.16,
  billTaxValue: '435435.16',
}, {
  name: '*锻件*通行费',
  modelNumber: 'ER602EGK6001',
  modelUnit: '个',
  modelCount: '20180108',
  modelPrice: '20180108',
  price: '3,000.00',
  taxRate: 0.16,
  billTaxValue: '435435.16',
}, {}]

const taxList = [{
  name: '餐饮费0',
  modelPrice: '300.00',
  modelCount: '847',
  sum: '1300'
}, {}]
const taxConfig = {
  billCode: '001',
  billNumber: '002',
  machineNumber: '003',
  machineId: '004',
  sellName: '山西溪唐小烹餐饮管理有限公司',
  sellTaxpayerNumber: '005',
  billDate: '006',
  receivingTeller: '007',
  buyName: '山西玉龙机电设备有限公司',
  buyTaxpayerNumber: '009',
  lowerSum: '-1600',
  upperSum: '101',
}
const taxConfigGS = {
  billCode: '001',
  billNumber: '002',
  industryType: '003',
  billDate: '004',
  machineCode: '005',
  machineId: '006',
  machineNumber: '007',
  taxControlCode: '008',
  payCompany: '009',
  upperSum: '010',
  billTaxPrice: '-7895',
  receivingCompany: '012',
  receivingCompanyTaxNumber: '013',
  remark: '014',
}
const taxListGS = [{
  name: '名称',
  modelNumber: '规格',
  modelUnit: '吨',
  modelCount: '100',
  modelPrice: '96.00',
  sum: '9,600.00'
}, {}]
const taxConfigJDC = {
  billCode: '001',
  billNumber: '002',
  billDate: '003',
  machineCode: '004',
  machineNumber: '005',
  machineId: '006',
  taxControlCode: 'wNr749ZfMwFjYxzAcpRRJNiYYaOR8liHt+cD5OkIDXuUC2f2DoxvMNTbXH0Vg7UCB/n418AVkt1bAmuAhAK43jBl92OL1biemixeo4sc5lbsFDWYFmatdPiSrRAoX6AmKMSmPh9q+V/Xa5w4wVzm9EeixtW6IJfMEhqJlwnH6rlzgfuJIk88e9CIhCT5gxZPG0z+tnVd80fAtZRhuF/eDqtrPwUzHWK2VDsx7U1ASbO4I2hu6e4iKJ0Vqyb5nD4',
  buyNameNumber: '008',
  buyTaxpayerNumber: '009',
  carType: '010',
  factoryPlateModel: '豪沃牌ZZ4257V324HE1B',
  producingArea: '012',
  qualifyCardNumber: '013',
  importCardNumber: '014',
  commodityInspectionNumber: '015',
  engineNumber: '016',
  vehicleIdentificationNumber: '017',
  billTaxPriceUpper: '018',
  billTaxPrice: '-1956',
  sellCompanyName: '020',
  sellCompanyAddrPhone: '021',
  sellTaxpayerNumber: '022',
  sellCompanyCode: '招商银行股份有限公司南京江宁支行 125909132810101',
  sellCompanyAddr: '024',
  sellCompanyBankNumber: '招商银行股份有限公司南京江宁支行 125909132810101',
  taxRate: '026',
  billTaxValue: '-2700',
  taxAuthoritiesCode: '国家税务总局南京市江宁区税务局 132011500',
  billNoTaxValue: '-2946',
  taxPaymentCardNumber: '030',
  tonnage: '031',
  limitedPeopleNumber: '032',
  otherMessage: {
    billNoTaxValue: '-2946',
    billTaxValue: '-2700',
    taxRateOrCharge: '0.06',
  }
}
const taxConfigDZ = {
  billNumber: '001',
  billDate: '002',
  billCode: '003',
  buyCompanyName: '004',
  buyTaxpayerNumber: '005',
  buy_addr_and_tel: '006',
  buy_bank_addr_and_account: '007',
  billTaxPriceUpper: '008',
  billTaxPrice: '-7894',
  sellCompanyName: '010',
  sellTaxpayerNumber: '011',
  sell_addr_and_tel: '012',
  sell_bank_addr_and_account: '013',
  remark: '01479797979879797979879879879879878979879878979797房间看得见经典款减肥福建跨境电商电视剧风口浪尖的大姐夫空间',
  make_type: 0,
  direction: 0,
  status_of_certification: 1,
  date_of_certification_date: '2020-01-09',
}
const taxListHG = [{}, {}]
const taxConfigHG = {}
const taxListWP = [{}, {}]
const taxConfigWP = {}

function render(container) {
  ReactDOM.render(
    <div>
      <div style={boxStyle}>
        {/* <Testone
          // taxConfig={taxConfigDZ}
          // taxDataList={taxDataList}
          // mode={'ZZSZYFP'}

          // mode={'ZZSPTFP'}
          
          // 电子普通发票
          // taxDataList={taxDataListB}
          // taxConfig={taxConfigDZ}
          // mode={'ZZSDZPTFP'}

          // mode={'ZZSDZTXF'}
          
          mode={'JDCXSTYFP'}
          prefixCls={'cr-tax-c'}
          taxConfig={taxConfigJDC}
        /> */}
        {/* <PTFPJP taxList={taxList} taxConfig={taxConfig} /> */}
        <GSTYJD taxList={taxListGS} taxConfig={taxConfigGS} />
        {/* <HGJKZZS taxList={taxListHG} taxConfig={taxConfigHG} /> */}
        {/* <WPSR taxList={taxListWP} taxConfig={taxConfigWP} /> */}
        {/* <ESCXS taxConfig={taxConfigJDC} /> */}
        {/* <LKYS taxDataList={taxDataListB} taxConfig={taxConfigDZ} /> */}
        {/* <NCP taxDataList={taxDataListB} taxConfig={taxConfigDZ} /> */}
      </div>
    </div>, container
  )
}

render(reactContainer)
